### PR TITLE
Block Editor: Update the packages to match Gutenberg 10.7.0

### PR DESCRIPTION
**Pull request by youknowriad**
Trac ticket: https://core.trac.wordpress.org/ticket/52991

## TODO

- [x] Remove the Post Author PHP code.
- [x] Update block supports.
- [x] Rename build_query_vars_from_query_block
- [x] Check PHP changes between 10.6 and 10.7. https://github.com/WordPress/gutenberg/compare/v10.6.0...v10.7.0-rc.1

### CODE REVIEW

File: src/wp-includes/block-supports/elements.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * Elements styles block support.
+ *
+ * @package WordPress
+ * @since 5.8.0
+ */
+
+/**
+ * Render the elements stylesheet.
+ *
```
Comment (gziolo) [inline src/wp-includes/block-supports/elements.php:11]: Nit:
```php
 * @since 5.8.0
 * @access private
```

**Comment by youknowriad** [comment]
I think I included all the mandatory php changes but would appreciate confirmation https://github.com/WordPress/gutenberg/compare/v10.6.0...v10.7.0-rc.1

**Comment by gziolo** [comment]
I assume that the changes for Block Patterns will be handled separately or do we keep the default ones in WP core for now?

File: src/wp-includes/block-supports/elements.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * @since 5.8.0
+ * @access private
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ * @return string                Filtered block content.
+ */
+function render_elements_support( $block_content, $block ) {
```
Comment (gziolo) [inline src/wp-includes/block-supports/elements.php:19]: In other files, we replaced `gutengerg_` with `wp_`.

**Comment by gziolo** [comment]
Two follow-ups required:
- Porting `gutenberg_extend_block_editor_styles_html` but I guess it's related to the template editor integration task
- Site Logo block needs to be integrated on the server together with changes to the REST API endpoint with settings (theme_mods and stylesheet changes).

**Comment by gziolo** [comment]
Committed with https://core.trac.wordpress.org/changeset/50945.

**Pull request closed without merging**